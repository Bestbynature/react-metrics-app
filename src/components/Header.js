import React from 'react';
import { MdOutlineArrowBackIosNew as Larr } from 'react-icons/md';
import { FaMicrophone as Mic } from 'react-icons/fa';
import { GoGear as Gear } from 'react-icons/go';
import { useNavigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { setHeader } from '../redux/details/detailsSlice';

const Header = () => {
  const navigate = useNavigate();

  const dispatch = useDispatch();

  const { header } = useSelector((store) => store.details);

  const { first, second } = header;

  const backHandler = () => {
    dispatch(setHeader());
    navigate('/', { replace: true });
  };

  return (
    <div className="header">
      <div><Larr onClick={backHandler} /></div>
      <h3>{first}</h3>
      <p>{second}</p>
      <Mic />
      <Gear />

    </div>
  );
};

export default Header;
