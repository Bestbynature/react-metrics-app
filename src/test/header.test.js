import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Header from '../components/Header';

const mockStore = configureStore([]);

describe('Header component', () => {
  test('renders Header component', () => {
    const initialState = {
      details: {
        header: {
          first: 'First Header',
          second: 'Second Header',
        },
      },
    };

    const store = mockStore(initialState);

    const { container, getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    );

    expect(container).toMatchSnapshot();

    const backButton = getByTestId('back-button');
    fireEvent.click(backButton);
  });
});
