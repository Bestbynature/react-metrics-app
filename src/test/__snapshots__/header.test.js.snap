// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header component 1`] = `
<div>
  <div
    class="header"
  >
    <div>
      <svg
        data-testid="back-button"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
        />
      </svg>
    </div>
    <h3>
      First Header
    </h3>
    <p>
      Second Header
    </p>
    <svg
      data-testid="mirophone-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 352 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
      />
    </svg>
    <svg
      data-testid="gear-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 14 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 8.77v-1.6l-1.94-.64-.45-1.09.88-1.84-1.13-1.13-1.81.91-1.09-.45-.69-1.92h-1.6l-.63 1.94-1.11.45-1.84-.88-1.13 1.13.91 1.81-.45 1.09L0 7.23v1.59l1.94.64.45 1.09-.88 1.84 1.13 1.13 1.81-.91 1.09.45.69 1.92h1.59l.63-1.94 1.11-.45 1.84.88 1.13-1.13-.92-1.81.47-1.09L14 8.75v.02zM7 11c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
